pulser-core/pulser/register/register.py

```diff
@@ -280,6 +286,7 @@ def rotate(self, degrees: float) -> None:
         Args:
             degrees: The angle of rotation in degrees.
         """
+        # TODO: Deprecate
```

Comment:
I don't like that there is a method that changes the Register in place, I think it's potentially problematic. It would be ok if this method returned a new register with the rotated coordinates though

Reply:
Let's create an issue for this